src/Omnipay/Common/Message/AbstractResponse.php

```diff
@@ -107,9 +107,24 @@ public function getCode()
         return null;
     }
 
+    /**
+     * Get the transaction reference as generated by the remote payment gateway.
+     *
+     * @return string
+     */
     public function getTransactionReference()
     {
-        return null;
```

Comment:
I'm guessing it was implemented and returned null to satisfy the interface, on the off chance that there was a gateway that didn't utilize Transaction Reference. Personally, I think if it's going to be part of the interface, it needs to return something meaningful in all cases. Otherwise, it shouldn't be part of the interface.